### PR TITLE
Remove required and description from CRD schema

### PR DESCRIPTION
**Pull request by tamalsaha**
xref: https://github.com/coreos/prometheus-operator/issues/1293
https://github.com/coreos/prometheus-operator/pull/1556#issuecomment-401799111

cc @brancz @ant31

### CODE REVIEW

File: pkg/crdvalidation.go [hunk 1]
```diff
@@ -40,9 +40,12 @@ func GetCustomResourceValidation(name string, fn func(ref common.ReferenceCallba
 	openapiSpec := fn(OpenAPIRefCallBack)
 	fixKnownTypes(openapiSpec)
 	schema := openapiSpec[name].Schema
-	return &extensionsobj.CustomResourceValidation{
+	crv := &extensionsobj.CustomResourceValidation{
 		OpenAPIV3Schema: SchemaPropsToJSONProps(&schema, openapiSpec, true),
 	}
+	crv.OpenAPIV3Schema.Description = ""
+	crv.OpenAPIV3Schema.Required = nil
```
Comment (ant31) [inline pkg/crdvalidation.go:47]: What is required ?

**Review by ant31 — APPROVED** [review]
LGTM

**Comment by brancz** [comment]
lgtm

File: pkg/crdvalidation.go (hunk 1 above)
Comment (tamalsaha) [reply pkg/crdvalidation.go:47]: It means required properties. This seems unsupported for top level properties in 1.10.

**Pull request merged**